### ISSUE TITLE
Benchmark Results From Ryzen 7 1700 1st Gen

### ISSUE THREAD

**Issue by davepoo** [cmuratori/meow_hash#63]
This is not an issue as such, but Casey mentioned on the HH stream the other day about testing Meow hash on AMD zen architecture. 

Although i can't send you my machine, I do own a 1st gen Ryzen 7 1700 https://en.wikichip.org/wiki/amd/ryzen_7/1700
So i've compiled and run the benchmarks using msvc 2017, and run the bench marks (windows 10),

Interesting to see the dropoff in speed at 8Mb, which matches with the 2x8mb of cache in the processor (i assume 8Mb per 4 core cluster), and at 32Kb (which is the L1 data cache size).

[results_ryzen7_1700_1stgen.zip](https://git
![meow_ryzen7_1700](https://user-images.githubusercontent.com/7448672/61986904-3a8d2180-b00a-11e9-8820-594a08d0788d.PNG)
hub.com/cmuratori/meow_hash/files/3437634/results_ryzen7_1700_1stgen.zip)

**Comment by cmuratori** [comment]
Do you have core clock boosting disabled (eg., "maximum CPU frequency" set to 99 instead of 100 in the Windows power settings)?  Because if you don't, then these results aren't really real.  If you _do_, though, holy cow - Ryzen eats i7/i9 for breakfast.  There's no Intel core chip that I've ever seen that can hit 16b/c on Meow 0.5.

But sadly I would suspect this is just speed boosting at work.

Either way, though, great to see that AMD chips run Meow very fast!

\- Casey

**Comment by davepoo** [comment]
> 
> Do you have core clock boosting disabled (eg., "maximum CPU frequency" set to 99 instead of 100 in the Windows power settings)? Because if you don't, then these results aren't really real.
> 

I had the windows power settings on best performance.
So I re-ran the benchmark with the 99% power, and i use the MSI mboard utility to monitor the CPU clocks while it was running. The clocks stayed at 3.2Ghz. The windows task manager was quoting values around 3.04 Ghz most of the time.
AMD say the base clock of the chip is 3.0 Ghz, and that it can turbo up to 3.7Ghz but i've never seen that. I am not doing any fancy overclock and am using the stock cooler that came with the processor. The MSI App says all the CPU cores are 3.2Ghz, so somehow that slightly above the base clock?

Either way, the result turned out pretty much the same.

[run_proc_99pc.zip](https://github.com/cmuratori/meow_hash/files/3438306/run_proc_99pc.zip)
![run_proc_99pc](https://user-images.githubusercontent.com/7448672/61996352-571d6e00-b08b-11e9-8264-47fceb8907ed.PNG)

**Comment by cmuratori** [comment]
This is really fascinating... I will have to get a Ryzen one of these days and investigate.  Naively, this means that the AMDs scheduler (or port config?) _doesn't_ have the limitation the i7/9 does that prevents them from hitting the theoretical maximum of 16 b/c.  That's awesome.  Actually I should go look and see if anyone has determined what the layout is for the Ryzens in terms of execution pipelines... maybe AES is a completely separate unit for them, or something.

\- Casey

**Comment by davepoo** [comment]
> 
> (That said, I suppose if RDTSC was running at 3.0gHz and the core was at 3.2, that would still account for the inflated score...)
> 
> - Casey

So i was wondering about this 3.2Ghz thing and why i've never seen this CPU boost.
I downloaded CPU-Z https://www.cpuid.com/softwares/cpu-z.html and (under tools -> clocks), it was showing that some of the cores were turboing to the 3.7Ghz as described in the specs (even when the windows power was set to 99 as i did before). Neither Window10 or the motherboards own software was showing that this was happening.

So the internet says i disable "Core Performance Boost" in the bios to get rid of the turbo, so i did that and low and behold - all the cores are now showing as 3.0Ghz (the correct clock for the CPU), and CPU-Z was not showing any turbo while moew was running (this is with the windows max CPU power on 100% instead of 99% as it didnt seem to do anything).

So to sum up:
* Setting windows power to "best performce" stops the throttling
* Setting the bios to "Core Performance Disabled" stops the boosting

So i ran meow again with the new "no throttle/ no boost" settings, and these are the new results, a little different this time.

![run_nothrottleorboost](https://user-images.githubusercontent.com/7448672/62005343-c77acc80-b129-11e9-97e5-d69109167082.PNG)
[run_nothrottleorboost.zip](https://github.com/cmuratori/meow_hash/files/3439056/run_nothrottleorboost.zip)

So looks like CPU-Z is the only way to reliably detect that turbo is happening, and all this explains why i never saw the CPU boost, and why i have a magic extra 0.2Ghz when the CPU is not overclocked.

I can now exit the rabbit hole.

**Comment by cmuratori** [comment]
OK _that_ is actually what I was expecting :)

\- Casey
